nodeup/pkg/model/kube_apiserver.go

```diff
@@ -216,12 +218,39 @@ func (b *KubeAPIServerBuilder) Build(c *fi.NodeupModelBuilderContext) error {
 	return nil
 }
 
+func (b *KubeAPIServerBuilder) configureOIDC(kubeAPIServer *kops.KubeAPIServerConfig) {
+	if b.NodeupConfig.APIServerConfig.Authentication == nil || b.NodeupConfig.APIServerConfig.Authentication.OIDC == nil {
```

Comment:
Could we do this on the kops-cli side?  The reason being that I expect the componentconfig objects eventually to become real component-config objects, and that we can create config files on the kops-cli side (also that they will come out of the cluster).  And in general, if we _can_ avoid logic in nodeup, I think we shoud.

But maybe there's a reason?

Reply:
It's the mindset. The admin should be thinking of configuring authentication to the cluster, not configuring the kube-apiserver binary. That OIDC is configured directly through kube-apiserver, whereas IAM and Kopeio are webhooks is an implementation detail that kOps should abstract away. As it is, it is difficult to discover how to configure OIDC because it is so buried.